presto-main/src/main/java/com/facebook/presto/sql/gen/BytecodeUtils.java

```diff
@@ -181,46 +179,92 @@ public static BytecodeNode generateInvocation(Scope scope, String name, ScalarFu
         // Index of parameter (without @IsNull) in Presto function
         int realParameterIndex = 0;
 
+        // go through all the choices in the function and then pick the best one
+        List<ScalarImplementationChoice> choices = function.getAllChoices();
+        int bestChoiceIndex = -1;
+        for (int i = 0; i < choices.size(); i++) {
+            boolean isInvalid = false;
+            ScalarImplementationChoice current = choices.get(i);
+            for (int j = 0; j < arguments.size(); j++) {
+                boolean isInputReference = false;
+                boolean isValueType = current.getArgumentProperty(j).getArgumentType() == VALUE_TYPE;
+                if (!isValueType) {
+                    continue;
+                }
+                if (arguments.get(j) instanceof InputReferenceNode) {
+                    isInputReference = true;
```

Comment:
Assignment to `isInputReference` can be simplified.

```
boolean isInputReference = arguments.get(j) instanceof InputReferenceNode;
```

In addition, you can inline it. In Presto, we generally prefer not having local variables that is only used once, unless there's compelling reasons.